hat/assets/js/apps/Iaso/domains/payments/hooks/requests/useSavePaymentLot.ts

```
@@ -0,0 +1,56 @@
+/* eslint-disable camelcase */
+import { UseMutationResult } from 'react-query';
+import { patchRequest, postRequest } from '../../../../libs/Api';
+import { useSnackMutation } from '../../../../libs/apiHooks';
+
+export type SavePaymentLotQuery = {
+    id?: number;
+    name: string;
+    comment?: string;
+    potential_payments?: number[];
+    mark_payments_as_sent?: boolean;
+};
+
+const endpoint = '/api/payments/lots/';
+
+const putPaymentLot = async (body: Partial<SavePaymentLotQuery>) => {
+    const url = `${endpoint}${body.id}/`;
+    const queryParams = body.mark_payments_as_sent
+        ? `?mark_payments_as_sent=${body.mark_payments_as_sent}`
+        : '';
+    return patchRequest(`${url}${queryParams}`, body);
+};
+
+const postPaymentLot = async (body: SavePaymentLotQuery) => {
+    return postRequest(endpoint, body);
+};
+
+export const useSavePaymentLot = (
+    type: 'create' | 'edit',
+    onSuccess?: () => void,
+): UseMutationResult => {
+    const ignoreErrorCodes = [400];
```

I don't think it's safe to ignore 400 error: the validation schema does not use the hook that handles backend errors, so I think it will just fail silently